### PR TITLE
Added Schema for User, Course and Reviews

### LINKED ISSUE #40: [Feature]: User schema
### Is your feature request related to a problem? Please describe.

Deals with the developer guide and the new schema #38

### Describe the solution you'd like

I am working on adding the new schema and the appropriate validations

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by divyanshu-parihar**
Closes #40

### CODE REVIEW

**Comment by divyanshu-parihar** [comment]
@YashKumarVerma  this pr is to add schema to the entities

**Comment by Aniket1118** [comment]
# op

File: src/modules/course/course.schema.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Define the structure of Course document here
+ */
+
+const courseSchema = new Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  instructor: {
```
Comment (YashKumarVerma) [inline src/modules/course/course.schema.js:15]: change to `instructors`

File: src/modules/course/course.schema.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    type: [{ type: Schema.Types.ObjectId, ref: 'User' }],
+  },
+  assistants: {
+    type: [{ type: Schema.Types.ObjectId, ref: 'User' }],
+  },
+  students: {
+    type: [{ type: Schema.Types.ObjectId, ref: 'User' }],
+  },
+  totalLikes: {
```
Comment (YashKumarVerma) [inline src/modules/course/course.schema.js:24]: change to `likes`

File: src/modules/review/review.schema.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    type: Date,
+  },
+  course: {
+    type: { type: Schema.Types.ObjectId, ref: 'Course' },
+  },
+  status: {
+    type: String,
+    enum: ['VISIBLE', 'FEATURED', 'BLOCKED'],
+  },
+  upvotes: {
```
Comment (YashKumarVerma) [inline src/modules/review/review.schema.js:27]: rename it to `likes`

**Review by YashKumarVerma — APPROVED** [review]

**Comment by YashKumarVerma** [comment]
quickly rename as suggested, then we're good to merge

**Comment by divyanshu-parihar** [comment]
@YashKumarVerma done

**Comment by divyanshu-parihar** [comment]
The user.routes files had lint error , i have changed that as that as well. 
lint error "line exceed character more than 120limit"

**Comment by divyanshu-parihar** [comment]
@YashKumarVerma 
 should i fix lint error and then make a commit or not?

**Review by YashKumarVerma — APPROVED** [review]

**Comment by YashKumarVerma** [comment]
please resolve the conflict, then we can directly merge it

**Comment by divyanshu-parihar** [comment]
@YashKumarVerma fixed merge errors

**Comment by YashKumarVerma** [comment]
thanks !

**Comment by divyanshu-parihar** [comment]
**nvm**

**Pull request merged**